statistics-dashboard-common/src/main/java/eu/europeana/statistics/dashboard/common/internal/RightsCategory.java

```diff
@@ -1,52 +1,93 @@
+            "^https?://www.europeana\\.eu/rights/unknown/.*$")),
+    UNKNOWN("Unknown", List.of());
 
     private final String name;
-    private final String regularExpression;
+    private final List<Pattern> regularExpressions;
 
-    RightsCategory(String name, String regularExpression){
+    RightsCategory(String name, List<String> regularExpressions) {
         this.name = name;
-        this.regularExpression = regularExpression;
+        this.regularExpressions = regularExpressions.isEmpty() ? new ArrayList<>() :
+                regularExpressions.stream().map(Pattern::compile).collect(Collectors.toList());
     }
 
     public String getName() {
         return name;
     }
 
-    public String getRegularExpression() {
-        return regularExpression;
+    public List<Pattern> getRegularExpressions() {
+        return regularExpressions;
     }
 
-    public static RightsCategory matchCategoryFromUrl(String url){
-        Set<RightsCategory> setRightsCategory = Arrays.stream(RightsCategory.values()).filter(x -> x != UNKNOWN).collect(Collectors.toSet());
+    /**
+     * Method that return a category that its regexes match the given url.
+     * Returns Unknown if the url matches none.
+     *
+     * @param url - The url to match to a category
+     * @return The category that its regexes match the given url. Unknown if matches none.
+     */
+    public static RightsCategory matchCategoryFromUrl(String url) {
+        Set<RightsCategory> setRightsCategory = Arrays.stream(RightsCategory.values()).filter(category -> category != UNKNOWN).collect(Collectors.toSet());
         RightsCategory result = UNKNOWN;
-        for(RightsCategory category: setRightsCategory){
-            if(url.matches(category.getRegularExpression())){
+        for (RightsCategory category : setRightsCategory) {
+            if (category.getRegularExpressions().stream().anyMatch(regex -> regex.matcher(url).find())) {
                 result = category;
                 break;
             }
         }
         return result;
     }
+
+    /**
+     * Method that returns the category that its name matches the given name.
+     *
+     * @param name The name to match to a category
+     * @return The category that its name match the given name
+     * @throws IllegalArgumentException if no such category with 'name' exists
+     */
+    public static RightsCategory toCategoryFromName(String name) throws IllegalArgumentException {
+        Set<RightsCategory> setRightsCategory = Arrays.stream(RightsCategory.values()).collect(Collectors.toSet());
+        RightsCategory result = null;
+        for (RightsCategory category : setRightsCategory) {
+            if (name.equals(category.getName())) {
+                result = category;
+                break;
+            }
+        }
+
+        if (result == null) {
+            throw new IllegalArgumentException("No such rights category with name '" + name + "' exists");
+        }
+        return result;
```

Comment:
I think this can be simpler and also better check case insensitively the names:
```suggestion
        for (RightsCategory category : RightsCategory.values()) {
          if (name.equalsIgnoreCase(category.getName())) {
            return category;
          }
        }
        throw new IllegalArgumentException(String.format("No such rights category with name '%s' exists", name);
```